javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -2586,6 +2586,17 @@ public static boolean isVarArgs(MethodInvocationTree invok) {
     return isVarArgs(elementFromUse(invok), invok.getArguments());
   }
 
+  /**
+   * Returns true if the given method invocation is a zero-argument varargs invocation.
+   *
+   * @param invok the method invocation
+   * @return true if the given method invocation is a zero-argument varargs invocation
+   */
+  public static boolean isZeroArgumentVarArgs(MethodInvocationTree invok) {
+    boolean isZeroArgumentCall = invok.getArguments().isEmpty();
+    return TreeUtils.isVarArgs(invok) && isZeroArgumentCall;
```

Comment:
For efficiency, don't compute the boolean that will be the second argument to the short-circuit `&&` before the first argument to `&&` is evaluated.

But, I'm not sure this method is needed; maybe just `isVarArgs()` would suffice.

Reply:
I think we still need a way to check whether a varargs parameter is empty, which is not something that we can know solely from the return value of `TreeUtils.isVarArgs`.

We need this information because the logic to handle qualifier replacement/completion is different depending on whether the varargs parameter is empty (or not). I might be misunderstanding something, though.